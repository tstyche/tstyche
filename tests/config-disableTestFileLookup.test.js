import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, writeFixture } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixture = "config-disableTestFileLookup";

afterEach(async () => {
  await clearFixture(fixture);
});

describe("'disableTestFileLookup' configuration file option", () => {
  test("does not search for test files", async () => {
    await writeFixture(fixture, {
      ["tstyche.config.json"]: JSON.stringify({ disableTestFileLookup: true }, null, 2),
    });

    const { exitCode, stderr, stdout } = await spawnTyche(fixture);

    expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
    expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

    expect(exitCode).toBe(0);
  });
});
