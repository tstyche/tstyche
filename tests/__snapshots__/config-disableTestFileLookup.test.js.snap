// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'disableTestFileLookup' configuration file option does not search for test files: stderr 1`] = `""`;

exports[`'disableTestFileLookup' configuration file option does not search for test files: stdout 1`] = `
"Targets:    1 passed, 1 total
Test files: 0 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
